File: src/sys/java/fan/sys/LogLevel.java
//
// Copyright (c) 2006, Brian Frank and Andy Frank
// Licensed under the Academic Free License version 3.0
//
// History:
//   3 Jul 06  Brian Frank  Creation
//
package fan.sys;

/**
 * LogLevel provides a set of discrete levels used to customize logging.
 */
public class LogLevel
  extends Enum
{

//////////////////////////////////////////////////////////////////////////
// Range
//////////////////////////////////////////////////////////////////////////

  public static final int DEBUG  = 0;
  public static final int INFO   = 1;
  public static final int WARN   = 2;
  public static final int ERR    = 3;
  public static final int SILENT = 4;

  public static final LogLevel debug  = new LogLevel(DEBUG,  "debug");
  public static final LogLevel info   = new LogLevel(INFO,   "info");
  public static final LogLevel warn   = new LogLevel(WARN,   "warn");
  public static final LogLevel err    = new LogLevel(ERR,    "err");
  public static final LogLevel silent = new LogLevel(SILENT, "silent");

  static final LogLevel[] array = { debug, info, warn, err, silent };

  public static final List<LogLevel> vals = (List)new List(Sys.LogLevelType, array).toImmutable();

//////////////////////////////////////////////////////////////////////////
// Implementation
//////////////////////////////////////////////////////////////////////////

  private LogLevel(int ord, String name)
  {
    Enum.make$(this, FanInt.pos[ord], name.intern());
    this.ord = ord;
  }

  public static LogLevel fromStr(String name) { return fromStr(name, true); }
  public static LogLevel fromStr(String name, boolean checked)
  {
    return (LogLevel)doFromStr(Sys.LogLevelType, name, checked);
  }

  public Type typeof() { return Sys.LogLevelType; }

  final int ord;

}

File: src/sys/dotnet/fan/sys/LogLevel.cs
//
// Copyright (c) 2006, Brian Frank and Andy Frank
// Licensed under the Academic Free License version 3.0
//
// History:
//   24 Dec 07  Andy Frank  Creation
//

namespace Fan.Sys
{
  /// <summary>
  /// LogLevel provides a set of discrete levels used to customize logging.
  /// </summary>
  public class LogLevel : Enum
  {

  //////////////////////////////////////////////////////////////////////////
  // Range
  //////////////////////////////////////////////////////////////////////////

    public static readonly LogLevel m_debug  = new LogLevel(0, "debug");
    public static readonly LogLevel m_info   = new LogLevel(1, "info");
    public static readonly LogLevel m_warn   = new LogLevel(2, "warn");
    public static readonly LogLevel m_err    = new LogLevel(3, "err");
    public static readonly LogLevel m_silent = new LogLevel(4, "silent");

    static readonly LogLevel[] array = { m_debug, m_info, m_warn, m_err, m_silent };

    public static readonly List m_vals = (List)new List(Sys.LogLevelType, array).toImmutable();

  //////////////////////////////////////////////////////////////////////////
  // Implementation
  //////////////////////////////////////////////////////////////////////////

    private LogLevel(int ord, string name)
    {
      Enum.make_(this, ord, System.String.Intern(name));
      this.m_ord = ord;
    }

    public static LogLevel fromStr(string name) { return fromStr(name, true); }
    public static LogLevel fromStr(string name, bool check)
    {
      return (LogLevel)doFromStr(Sys.LogLevelType, name, check);
    }

    public override Type @typeof() { return Sys.LogLevelType; }

    internal readonly int m_ord;

  }
}